### DISCUSSION TITLE
How to make a button execute a action when pressed?

### DISCUSSION THREAD

**Discussion by pparaxan** [slint-ui/slint#3438] [Q&A] [answered]
You could say I'm "new" to Rust & Slint, I've started using it for around a weeks now.

**Comment by pparaxan** [comment]
I want a button to execute a rust command
(.slint)

**Comment by Vadoola** [comment, marked as answer]
What you are probably looking for is the clicked callback on the button. Here is an example of basic usage withing slint in the docs: https://slint.dev/releases/1.1.1/docs/slint/src/recipes/recipes.

For doing something from the Rust code when a button is clicked you would want to handle the on_mybutton_clicked callback in the rust code.


```
let btn_handle = main.as_weak();
    main.on_minimize_window(move || {
        let v= btn_handle .upgrade().unwrap();
        //logic on button pressed here
    });
});
```

You could see some more examples in my code [https://github.com/Vadoola/Tomotroid/blob/main/src/main.rs](here: https://github.com/Vadoola/Tomotroid/blob/main/src/main.rs)


or look at the  main_window.on_filter_image section of the [Slint ImageFilter Demo](https://github.com/slint-ui/slint/blob/master/examples/imagefilter/main.rs)

**Comment by pparaxan** [comment]
Thanks!

**Reply by Vadoola** [reply]
Happy to help
